src/Content/MailArchive/MailArchiveDefinition.php

```diff
@@ -47,6 +56,7 @@ protected function defineFields(): FieldCollection
             (new LongTextField('eml', 'eml'))->addFlags(new AllowHtml()),
             (new StringField('eml_path', 'emlPath', 2048)),
             (new StringField('transport_state', 'transportState'))->addFlags(new Required()),
+            new BoolField('history_last_mail', 'historyLastMail'),
```

Comment:
I don't really like the idea of adding a separate column for `historyLastMail`. Why not just do a groupBy on `sourceMailId`/`parentId`?

Reply:
@schneider-felix 
As discussed on Saturday, I tried a GroupBy via a historyGroupId column.
This column can be realized as a virtual column that reflects the value of parent_id/id.

However, this does not work as expected.
When I use criteria.addGrouping('historyGroupId') in the admin, the result is not correct.
The last element is not returned as expected.

Did I use the addGrouping incorrectly or does it not work like a GroupBy in SQL?
The customized code can be found at: [Link to Branch](https://github.com/flkasper/FroshPlatformMailArchive/compare/feature/filter-last-history-mail-transport-state...flkasper:FroshPlatformMailArchive:feature/filter-last-history-mail-transport-state-test?expand=1)

If we cannot fix the grouping, we will have to stick with the less attractive solution with historyLastMail.

Reply:
@flkasper thanks for the update. I'll have a look.